src/ethereum_spec_tools/diff.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+"""
+Generates diffs between Ethereum hardforks documentation.
+"""
+
+import os.path
+import pickle
+from typing import Iterator, List, TypeVar
+
+import rstdiff
+from docutils import SettingsSpec
+from docutils.utils import new_document, new_reporter
+
+from ethereum_spec_tools.forks import Hardfork
+
+T = TypeVar("T")
+
+
+def window(forks: List[T], window_size: int = 2) -> Iterator[List[T]]:
+    """
+    Group a list into overlapping chunks.
+    """
+    for i in range(len(forks) - window_size + 1):
+        yield forks[i : i + window_size]
+
+
+def find_pickles(path: str, fork: Hardfork) -> Iterator[str]:
+    """
+    Find files ending with :code:`.pickle`.
+    """
+    for directory, _, filenames in os.walk(path):
+        for filename in filenames:
+            if filename.endswith(".pickle"):
+                file_path = os.path.join(directory, filename)
+                file_path = file_path[len(path) + 1 :]
+                yield file_path
+
+
+def diff(
+    output_path: str, input_path: str, old: Hardfork, new: Hardfork
+) -> Iterator[str]:
+    """
+    Calculate the structured diff between two hardforks.
+    """
+    old_path = old.name.replace(".", os.sep)
+    old_path = os.path.join(input_path, old_path)
+
+    new_path = new.name.replace(".", os.sep)
+    new_path = os.path.join(input_path, new_path)
+
+    diff_path = old.name.split(".")[-1] + "_" + new.name.split(".")[-1]
+    diff_path = os.path.join(output_path, diff_path)
+
+    old_pickles = set(find_pickles(old_path, old))
```

Comment:
Just curious, why return an iterator and create a set vs. simply returning a set? More flexibility?

Reply:
Didn't really think much about it. Seems a bit clunky to write:

```python
x = set()
for y in z:
    x.append(x)
return x
```

Reply:
okay that makes sense, you're version is a lot more terse